packages/grid/theme/lumo/vaadin-grid-styles.js

```diff
@@ -309,8 +309,8 @@ registerStyles(
 
     /* Row stripes */
 
-    :host([theme~='row-stripes']) [part~='row']:not([odd]) [part~='body-cell'],
-    :host([theme~='row-stripes']) [part~='row']:not([odd]) [part~='details-cell'] {
+    :host([theme~='row-stripes']) [part~='row']:not([part~='odd-row']) [part~='body-cell'],
+    :host([theme~='row-stripes']) [part~='row']:not([part~='odd-row']) [part~='details-cell'] {
```

Comment:
Could we reverse the part name here?

```suggestion
    :host([theme~='row-stripes']) [part~='row'][part~='even-row'] [part~='body-cell'],
    :host([theme~='row-stripes']) [part~='row'][part~='even-row'] [part~='details-cell'] {
```